framework/src/binary_manager/binary_manager_get_state.c

```diff
@@ -0,0 +1,65 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/***************************************************************************
+ * Included Files
+ ***************************************************************************/
+
+#include <debug.h>
+#include <string.h>
+#include <tinyara/binary_manager.h>
+#include <binary_manager/binary_manager.h>
+#include "binary_manager_internal.h"
+
+int binary_manager_get_state(char *binary_name)
+{
+	int ret;
+	binmgr_request_t request_msg;
+	binmgr_getstate_response_t response_msg;
+
+	if (binary_name == NULL || strlen(binary_name) > BIN_NAME_MAX - 1) {
+		bmdbg("get_binary_info failed : invalid param.\n");
+		return BINMGR_INVALID_PARAM;
+	}
+
+	ret = binary_manager_set_request(&request_msg, BINMGR_GET_STATE, binary_name);
```

Comment:
binary_manager_set_request should return BINMGR_COMMUNICATION_FAIL or something, instead of ERROR to reduce unnecessary code here.
Actually line 45 in binary_manager_set_request should return BINMGR_INVALID_PARAM or something. of course line 35 in this function checking it but it is necessary if there's chance that some other place also call binary_manager_set_request.